addons/crm/static/src/js/systray_activity_menu.js

```diff
@@ -2,25 +2,64 @@ odoo.define('crm.systray.ActivityMenu', function (require) {
 "use strict";
 
 var ActivityMenu = require('mail.systray.ActivityMenu');
-
+var session = require('web.session');
 ActivityMenu.include({
-
     //--------------------------------------------------
     // Private
     //--------------------------------------------------
-
     /**
      * @override
      */
     _getViewsList(model) {
         if (model === "crm.lead") {
-                return [[false, 'kanban'], [false, 'list'],
-                        [false, 'form'], [false, 'calendar'],
-                        [false, 'pivot'], [false, 'graph'],
-                        [false, 'activity']
-                    ];
+            return [[false, 'list'], [false, 'kanban'],
+                    [false, 'form'],[false, 'calendar'],
+                    [false, 'pivot'], [false, 'graph'],
+                    [false, 'activity']
+                ];
         }
         return this._super(...arguments);
     },
+
+    /**
+     * Redirect to particular model view
+     * @private
+     * @param {MouseEvent} event
+     */
+    _onActivityFilterClick: function (event) {
+        // fetch the data from the button otherwise fetch the ones from the parent (.o_mail_preview).
+        var action;
+        var data = _.extend({}, $(event.currentTarget).data(), $(event.target).data());
+        var additional_context = {};
+
+        if (data.model_name === 'Lead'){
```

Comment:
Not sure where model_name comes from, but it seems like it is some kind of string based on server data, that could be translated ... Shouldn't you propagate a type (lead_type) ?